### PR TITLE
mednafen 0.9.38.6

### PR DESCRIPTION
**Pull request by tomyun**
As @yisraeldov worked on 0.9.38.5 with #330, so not sure if it's okay to open another PR, but here is a formula for the latest version, 0.9.38.6, with some other updates.

- version bump
- `fails_with` added
  - prevents test failure with LLVM (up to) 3.6
  - needs `gcc` package (do we need hard dep for `:build`?)
- Dependency update
  - `libcdio` removed (physical CD support dropped since 0.9.38)
  - `gettext` added for locale support
- `ENV.cxx11` added for Mountain Lion
- `test` block made up with `-dump_modules_def` feature.

### CODE REVIEW

**Comment by yisraeldov** [comment]
Mine is not working on all versions of osx .

**Comment by yisraeldov** [comment]
Why did you remove libcdio?

**Comment by tomyun** [comment]
`libcdio` was used for providing physical CD support and it's been removed since 0.9.38 according to the [announcement](http://forum.fobby.net/index.php?t=rview&goto=3874&th=1145). It's no longer listed in the [dependencies](http://mednafen.fobby.net/releases/).

I also confirmed with `otool` that the final binary didn't link to it anyways.

**Comment by yisraeldov** [comment]
hi

this didn't build for me 

```
==> Installing homebrew/games/mednafen
Warning: Formula file is modified!
Building from source because Library/Taps/homebrew/homebrew-games/mednafen.rb has local changes
To install from a bottle instead, run with --force-bottle
couldn't understand kern.osversion `14.4.0'
==> Downloading http://mednafen.fobby.net/releases/files/mednafen-0.9.38.6.tar.bz2
/usr/bin/curl -fLA Homebrew 0.9.5 (Ruby 2.0.0-481; OS X 10.10.4) http://mednafen.fobby.net/releases/files/mednafen-0.9.38.6.tar.bz2 -C 0 -o /Library/Caches/Homebrew/mednafen-0.9.38.6.tar.bz2.incomplete
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3804k  100 3804k    0     0   229k      0  0:00:16  0:00:16 --:--:--  187k
==> Verifying mednafen-0.9.38.6.tar.bz2 checksum
tar xf /Library/Caches/Homebrew/mednafen-0.9.38.6.tar.bz2
ln -s ../Cellar/mednafen/0.9.38.5/bin/mednafen mednafen
ln -s ../../../../Cellar/mednafen/0.9.38.5/share/locale/de/LC_MESSAGES/mednafen.mo mednafen.mo
ln -s ../../../../Cellar/mednafen/0.9.38.5/share/locale/es/LC_MESSAGES/mednafen.mo mednafen.mo
ln -s ../../../../Cellar/mednafen/0.9.38.5/share/locale/ru/LC_MESSAGES/mednafen.mo mednafen.mo
ln -s ../Cellar/mednafen/0.9.38.5/share/mednafen mednafen
Error: The selected compiler doesn't support C++11: gcc-4.2
```

using built in gcc

File: mednafen.rb [hunk 1]
```diff
@@ -11,13 +10,34 @@ class Mednafen < Formula
...
+    cause <<-EOS.undent
+      LLVM miscompiles some loop code with optimization
+      https://llvm.org/bugs/show_bug.cgi?id=15470
+      EOS
+  end
+
+  fails_with :llvm do
+    build 2336
+    cause <<-EOS.undent
+      llvm-gcc does not support C++11
```
Comment (mistydemeo) [inline mednafen.rb]: You can use `needs :cxx11` to set all of the appropriate failures aside from clang; it'll also do gcc-4.2, etc.

Comment (tomyun) [inline mednafen.rb]: Thanks! So do we need both `needs :cxx11` and `ENV.cxx11` in such formulae with hard dependency on C++11?

Comment (mistydemeo) [inline mednafen.rb]: Yep! `needs :cxx11` sets up the list of compatible compilers, while `ENV.cxx11` adds the appropriate flags to build using C++11 during the build.

**Comment by mistydemeo** [comment]
> needs gcc package (do we need hard dep for :build?)

Homebrew will prompt the user to install it if they don't have a compatible compiler, this is fine.

**Pull request closed without merging**